Emulator/BF.File.Emulator/Bf/BfBuilder.cs

```diff
@@ -0,0 +1,175 @@
+﻿using FileEmulationFramework.Lib.IO;
+using FileEmulationFramework.Lib.Utilities;
+using FileEmulationFramework.Lib.Memory;
+using AtlusScriptLibrary.FlowScriptLanguage.Compiler;
+using FlowFormatVersion = AtlusScriptLibrary.FlowScriptLanguage.FormatVersion;
+using AtlusScriptLibrary.Common.Libraries;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+using AtlusScriptLibrary.FlowScriptLanguage;
+using Newtonsoft.Json;
```

Comment:
Would you be able to fork `AtlusScriptLibrary` and port it to use `System.Text.Json`?
The usage of Newtonsoft there is very minimal, so it shouldn't be too hard.

There are two main reasons:

## 1. Efficiency/File Size Concerns

Newtonsoft is a 700KB binary that is not safe for trimming, and adds considerable time to startup due to the need to JIT compile a lot of code from scratch. 

I don't remember the exact numbers but I recall it being in the range of 40-100ms extra on startup on a mid range CPU. This is a lot.

## 2. This will ease the Rust transition for you.

We can't use C# code in OS API hooks, due to GC, unfortunately.

However, there *is* a workaround, but it's costly performance wise; however I'm happy to allow it in an initial port while actual `AtlusScriptLibrary` is ported.

(Note: When you get around to that, don't port the whole library if you don't want to; just the stuff you'd need, walk through the APIs and port all the actually used code)

The workaround is to run the C# code in a separate thread (i.e. run the code you need in another thread, and pass back to your current thread).

You could achieve that with either:
- Building [NativeAOT](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary) library and dynamically linking.
- Dependency Injection. 

    Soon-ish in R2 and R3, mods will be able to export interfaces that can be consumed from native code (the same way you can between C# mods, albeit with slightly more limitations since you're limited to basic C types and memory management), and you could invoke those APIs from a separate thread.

Technically only the NativeAOT solution requires `System.Text.Json` such that trimming is possible; however, I still find Newtonsoft to be too heavy for the purpose.

Reply:
Done in 9928c2f (and the [related commit](https://github.com/AnimatedSwine37/Atlus-Script-Tools/commit/5e576a058128e5c171a82e2f3930338ce62bd2ec) in my script compiler fork)

Reply:
I left a comment in [script compiler](https://github.com/AnimatedSwine37/Atlus-Script-Tools/commit/5e576a058128e5c171a82e2f3930338ce62bd2ec#r120339745), you still need to conditionally exclude the dependency from runtimes which already ship it; otherwise it won't use the runtime's version.

Reply:
Thanks for pointing that out, I've done that now ([commit in script tools](https://github.com/AnimatedSwine37/Atlus-Script-Tools/commit/67ac9172181fb9f526b674fc6af0f74c7dcdf0e7))